src/services/completions.ts

```diff
@@ -1342,9 +1343,12 @@ namespace ts.Completions {
                     }
 
                     const symbolId = getSymbolId(symbol);
-                    symbols.push(symbol);
-                    symbolToOriginInfoMap[symbolId] = origin;
-                    symbolToSortTextMap[symbolId] = SortText.AutoImportSuggestions;
+                    const existingSymbol = findLast(symbols, symbol => symbol.id === symbolId);

```

Comment:
There’s existing deduplication logic in `getSymbolsFromOtherSourceFileExports` which IIRC maintains O(n) time complexity. I worry about introducing O(n^2) here—hopefully the logic in there can just be tweaked?

Reply:
Perfect, yes!

Reply:
Yeah, check the condition by the comment that says

```ts
// Don't add another completion for `export =` of a symbol that's already global.
// So in `declare namespace foo {} declare module "foo" { export = foo; }`, there will just be the global completion for `foo`.
```

The check that it’s doing isn’t good enough—just because the file is a module doesn’t mean the symbol can’t be global. It seems it doesn’t account for `declare global` _or_ `export as namespace`.